smack-websocket/src/main/java/org/jivesoftware/smack/websocket/elements/OpenElement.java

```diff
@@ -0,0 +1,47 @@
+/**
+ *
+ * Copyright 2020 Aditya Borikar
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smack.websocket.elements;
+
+import javax.xml.namespace.QName;
+
+import org.jivesoftware.smack.packet.StreamOpen;
+import org.jivesoftware.smack.packet.XmlEnvironment;
+
+import org.jxmpp.jid.DomainBareJid;
+
+public final class OpenElement extends StreamOpen {
+    public static final String ELEMENT = "open";
+    public static final String NAMESPACE = "urn:ietf:params:xml:ns:xmpp-framing";
+    public static final QName QNAME = new QName(NAMESPACE, ELEMENT);
+
+    public OpenElement(DomainBareJid to) {
+        super(to);
+    }
+
+    @Override
+    public String getElementName() {
+        return ELEMENT;
+    }
+
+    @Override
+    public CharSequence toXML() {
+        XmlEnvironment enclosingEnvironment = new XmlEnvironment(NAMESPACE);
```

Comment:
The whole approach to this is not good. Don't try to wildly patch things until the look like what you want.

Instead you create a method in `AbstractStreamOpen` that is called `addCommonAttributes(XmlStringBuilder)` that is used in the toXML() method of `StreamOpen` and here.